python/pyarrow/error.pxi

```diff
@@ -24,6 +24,9 @@ from contextlib import contextmanager
 import os
 import signal
 import threading
+import weakref
```

Comment:
Do you need weakref here?

Reply:
No, it's just a remnant from a failed attempt.